department-of-veterans-affairs/caseflow
# Ki/APPEALS-36418 - Autotext Modal Updates

Resolves [Autotexts Modal](https://jira.devops.va.gov/browse/APPEALS-36418)

# Description
- Added a "Clear all" button to autotexts modal
- Updated logic for continually adding text and autotext for tasks
- Updated and Created new RSpec tests

## Acceptance Criteria

## Testing Plan
1. Go to Step 2 of the Intake Form
2. To get the Task Modal to appear, either:
     1. Click "+Add tasks" button under _Tasks not related to an Appeal_
     2. Click "Yes" under _Tasks related to an existing Appeal_
         1. Select an appeal from Existing Appeals list and scroll down
3. Test the Updated "Add" functionality for the Context/Content box for text
    1. Enter input into the context box -- either Manual text input, or using the "Add autotext option"
    2. Any new selections from the Autotext Modal will now be appended to the end of the content box
4. Test new "Clear all" function
    1. Select any options in Autotext Modal
    2. Click "Clear all"
    3. Should unselect any(all) options that were previously selected.  Also returns modal "Add" option to greyed-out state.

# Frontend
## User Facing Changes

### Clear All Function
![Screenshot 2023-12-13 at 3 56 41 PM](https://github.com/department-of-veterans-affairs/caseflow/assets/108023343/ab33f30a-70f9-4ce9-8cf9-efae887377c9)
![Screenshot 2023-12-13 at 3 56 46 PM](https://github.com/department-of-veterans-affairs/caseflow/assets/108023343/ff0ac64a-1f1b-44be-aa6b-f0983bb2cf9b)

### Add (Appending Text) Function
![Screenshot 2023-12-13 at 3 57 20 PM](https://github.com/department-of-veterans-affairs/caseflow/assets/108023343/e88bc087-0bff-4562-be3e-ee4a6502fc55)
![Screenshot 2023-12-13 at 3 57 28 PM](https://github.com/department-of-veterans-affairs/caseflow/assets/108023343/0765e2d1-ff3e-4aad-8a14-903ab3b623a8)
![Screenshot 2023-12-13 at 3 57 33 PM](https://github.com/department-of-veterans-affairs/caseflow/assets/108023343/3107efb9-0f54-4f9c-bf1c-b894081f69b0)
![Screenshot 2023-12-13 at 3 58 10 PM](https://github.com/department-of-veterans-affairs/caseflow/assets/108023343/3ffceaae-16c7-451a-981f-7a2c9a5d9eac)

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:
